2016/proposals/verma-nikhil.md

```diff
@@ -0,0 +1,232 @@
+
+   * Individual details of every member, as in [1] above
+   * Group Leader
+   * Whether or not they are willing to expand the group
+
+
+### 2. Managing Trainees - The Trainee Admin
+
+  The Trainee Admin will be used to manage trainees at each stage of training. Access will be based on permissions (to be discussed during GSoC), so that the right people get to change and monitor the trainee status at each stage. For now, an *admin* is the relevant person handling a particular stage for an applicant. The admin workflow is outlined here:
+
+  * All the applications will be visible to *admins* on a dashboard.
+  * The *admins* start by reviewing applications for instructor training that were filled in the form above.
+  * They can then shortlist applicants and match them to courses. Multiple applications can be updated at once for ease.
+  * Once this is done, the applicants are sent an invitation to attend the selected course and also to join the Trainee Dashboard.
+
+  >Trainees log in, and attend the course
+
+  * The *admin* updates course completion status
+
+  >Trainees submit homework link
+
+  * The *admin* reviews the homework, and provide necessary feedback. Once they are satisfied, they update homework completion status.
+
+  >Trainee provides preferred time for discussion session (optional)
+
+  * The *admin* fixes up a discussion session
+
+  >Trainees attend discussion session
+
+  * The *admin* evaluates the discussion session, providing necessary feedback. If satisfied, updates the discussion complete status
+
+  >Trainee provides preferred time for discussion session (optional)
+
+  * The *admin* fixes up a checkout session
+
+  >Trainees attend checkout session
+
+  * The *admin* evaluates the checkout session, providing necessary feedback. If satisfied, updates the checkout complete status
+  * When this is done, the applicant now becomes a trainee, and they can be managed from the AMY instructor dashboard.
+
+
+### 3. Trainee Dashboard
+
+  Every applicant that gets selected will be provided with an access to the trainee dashboard. This will be a one stop solution to all their queries. Most importantly, they'll know what has been done and what to do next. Here, the trainee can track their progress, interact with instructors at any stage, and find out the next steps need to be done. This makes things simpler for everyone.
+
+  * **Know alloted course**
+
+  As soon as the trainee logs in, they'll be shown with the course they are matched with. When they complete the course, they'll be taken to the next step - homework submission.
+
+  * **Homework Submission**
+
+  Trainees can submit the link to their pull request or Google Doc. This way it'll be easier to track submissions.
+
+  * **Discussion sessions**
+
+  The trainees will have an option to select a time slot out of the available ones. This makes matching easier.
```

Comment:
I suppose the Google Calendar would be updated only when the discussion leader actually wants to update slots. I mean, there's no need to *confirm* the change *again* from the discussion leader. 

Also, I don't think there's any way by which we can get to know whether an event is changed as soon as that happens - which means yes, we'll have to ping to the specified calendar from time to time (a cron job).

Now when the cron script detects a change, it can apply it immediately and notify the concerned people via email. These would be the software admin (person handling amy), and the discussion leader. The trainees may also want to act accordingly, so I guess they should also be notified.